.github/workflows/deploy.yml

```diff
@@ -116,6 +116,24 @@ jobs:
 
       - name: Deploy infrastructure to UAT
         run: kubectl apply --record=false -f ./infrastructure/laa-manage-your-civil-cases-uat/
+      
+      - name: Set environment variables
```

Comment:
Because you are repeating yourself in three different job, it makes more sense to use Github Composite Action for these set environment variables sections.

Break it out into it's own yml and reference the yml file in the jobs

There are other ways to re-use these actions in github